### DISCUSSION TITLE
Audit log streaming API requests targeting Enterprises' private repositories - [Beta Feedback]

### DISCUSSION THREAD

**Discussion by boylejj** [community/community#45920] [Enterprise]
# Public Beta now available!

## Summary

GitHub Enterprise Cloud customers can now participate in a private beta allowing API requests targeting private repositories to be streamed to one of [GitHub's supported streaming endpoints](https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/streaming-the-audit-log-for-your-enterprise). Many GitHub users leverage GitHub's application programming interfaces (APIs) to extend and customize their GitHub experience. However, use of APIs can create unique security and operational challenges for Enterprises.

## Intended Outcome

With the introduction of audit log streaming API requests targeting Enterprises' private repositories, Enterprise owners are able to:
* Better understand and analyze API usage targeting their private repositories
* Identify and diagnose potentially misconfigured applications or integrations
* Troubleshoot API activity targeting private repositories that may be contributing to API rate limiting
* Develop API specific anomaly detection algorithms to identify potentially malicious activity

## How will it work?
Enterprise owners interested in participating in the private beta should reach out to your GitHub account manager or [contact our sales team](https://github.com/enterprise/contact) to have this feature enabled for your enterprise. Once enabled, you should begin seeing API request events in your audit log stream.

## Routes Selected
<details>
  <summary>Enterprise </summary>

- /enterprise-installation
- /enterprise-installation/:enterprise_or_org/server-statistics
- /enterprise-installation/application
- /enterprise-installation/contributions
- /enterprise-installation/permissions
- /enterprise-installation/usage_metrics
- /enterprise-installation/usage-metrics
- /enterprise-installation/vulnerabilities
- /enterprises/:enterprise_id/:security_feature/:enablement
- /enterprises/:enterprise_id/code_security_and_analysis
- /enterprises/:enterprise_id/secret-scanning/alerts
- /enterprises/:enterprise_id/settings/billing/actions
- /enterprises/:enterprise_id/settings/billing/advanced-security
- /enterprises/:enterprise_id/settings/billing/packages
- /enterprises/:enterprise_id/settings/billing/shared-storage

</details>

<details>
  <summary>Integrations</summary>

- /installations/:installation_id/repositories/:repository_id

</details>

<details>
  <summary>GraphQL</summary>

- /graphql

</details>

<details>
  <summary>Organizations</summary>

- /organizations
- /organizations/:org_id/team/:team_id
- /organizations/:org_id/team/:team_id/external-groups
- /organizations/:org_id/team/:team_id/invitations
- /organizations/:org_id/team/:team_id/members
- /organizations/:org_id/team/:team_id/members/:username
- /organizations/:org_id/team/:team_id/memberships/:username
- /organizations/:org_id/team/:team_id/repos
- /organizations/:org_id/team/:team_id/repositories/:repository_id
- /organizations/:org_id/team/:team_id/team-sync/group-mappings
- /organizations/:org_id/team/:team_id/team-sync/group-mappings-state
- /organizations/:org_id/team/:team_id/teams
- /organizations/:organization_id
- /organizations/:organization_id/:security_feature/:enablement
- /organizations/:organization_id/actions/permissions
- /organizations/:organization_id/actions/permissions/repositories
- /organizations/:organization_id/actions/permissions/repositories/:repository_id
- /organizations/:organization_id/actions/permissions/selected-actions
- /organizations/:organization_id/actions/permissions/workflow
- /organizations/:organization_id/actions/required_workflows
- /organizations/:organization_id/actions/required_workflows/:required_workflow_id
- /organizations/:organization_id/actions/required_workflows/:required_workflow_id/repositories
- /organizations/:organization_id/actions/required_workflows/:required_workflow_id/repositories/:repository_id
- /organizations/:organization_id/actions/secrets
- /organizations/:organization_id/actions/secrets/:name
- /organizations/:organization_id/actions/secrets/:name/repositories
- /organizations/:organization_id/actions/secrets/:name/repositories/:repository_id
- /organizations/:organization_id/actions/secrets/public-key
- /organizations/:organization_id/actions/variables
- /organizations/:organization_id/actions/variables/:name
- /organizations/:organization_id/actions/variables/:name/repositories
- /organizations/:organization_id/actions/variables/:name/repositories/:repository_id
- /organizations/:organization_id/blocks
- /organizations/:organization_id/blocks/:username
- /organizations/:organization_id/codespaces/secrets
- /organizations/:organization_id/codespaces/secrets/:name
- /organizations/:organization_id/codespaces/secrets/:name/repositories
- /organizations/:organization_id/codespaces/secrets/:name/repositories/:repository_id
- /organizations/:organization_id/codespaces/secrets/public-key
- /organizations/:organization_id/credential-authorizations
- /organizations/:organization_id/credential-authorizations/:credential_auth_id
- /organizations/:organization_id/custom_roles
- /organizations/:organization_id/custom_roles/:role_id
- /organizations/:organization_id/custom-organization-roles
- /organizations/:organization_id/custom-organization-roles/:role_id
- /organizations/:organization_id/custom-organization-roles/team/:team_id/:role_id
- /organizations/:organization_id/custom-organization-roles/users/:username/:role_id
- /organizations/:organization_id/custom-repository-roles
- /organizations/:organization_id/custom-repository-roles/:role_id
- /organizations/:organization_id/dependabot/secrets
- /organizations/:organization_id/dependabot/secrets/:name
- /organizations/:organization_id/dependabot/secrets/:name/repositories
- /organizations/:organization_id/dependabot/secrets/:name/repositories/:repository_id
- /organizations/:organization_id/dependabot/secrets/public-key
- /organizations/:organization_id/external-group/:group_id
- /organizations/:organization_id/external-groups
- /organizations/:organization_id/failed_invitations
- /organizations/:organization_id/hooks
- /organizations/:organization_id/hooks/:hook_id
- /organizations/:organization_id/hooks/:hook_id/config
- /organizations/:organization_id/hooks/:hook_id/deliveries
- /organizations/:organization_id/hooks/:hook_id/deliveries/:delivery_id
- /organizations/:organization_id/hooks/:hook_id/deliveries/:delivery_id/attempts
- /organizations/:organization_id/hooks/:hook_id/pings
- /organizations/:organization_id/installation
- /organizations/:organization_id/installations
- /organizations/:organization_id/invitations
- /organizations/:organization_id/invitations/:invitation_id
- /organizations/:organization_id/invitations/:invitation_id/teams
- /organizations/:organization_id/organization-fine-grained-permissions
- /organizations/:organization_id/outside_collaborators
- /organizations/:organization_id/outside_collaborators/:username
- /organizations/:organization_id/personal-access-token-requests
- /organizations/:organization_id/personal-access-token-requests/:pat_request_id
- /organizations/:organization_id/personal-access-token-requests/:pat_request_id/repositories
- /organizations/:organization_id/personal-access-tokens
- /organizations/:organization_id/personal-access-tokens/:pat_id
- /organizations/:organization_id/personal-access-tokens/:pat_id/repositories
- /organizations/:organization_id/security-managers
- /organizations/:organization_id/security-managers/team/:team_id
- /organizations/:organization_id/settings/billing/actions
- /organizations/:organization_id/settings/billing/advanced-security
- /organizations/:organization_id/settings/billing/packages
- /organizations/:organization_id/settings/billing/shared-storage
- /organizations/:organization_id/team-sync/groups
- /organizations/:organization_id/team/:team_id
- /organizations/:organization_id/teams

</details>

<details>
  <summary>Repositories</summary>

- /repositories/:repository_id/actions/organization-secrets
- /repositories/:repository_id/actions/organization-variables
- /repositories/:repository_id/actions/permissions
- /repositories/:repository_id/actions/permissions/access
- /repositories/:repository_id/actions/permissions/selected-actions
- /repositories/:repository_id/actions/permissions/workflow
- /repositories/:repository_id/actions/required_workflows
- /repositories/:repository_id/actions/required_workflows/:required_workflow_id
- /repositories/:repository_id/actions/required_workflows/:required_workflow_id/timing
- /repositories/:repository_id/actions/secrets
- /repositories/:repository_id/actions/secrets/:name
- /repositories/:repository_id/actions/secrets/public-key
- /repositories/:repository_id/actions/variables
- /repositories/:repository_id/actions/variables/:name
- /repositories/:repository_id/branches
- /repositories/:repository_id/branches/*
- /repositories/:repository_id/branches/*/protection
- /repositories/:repository_id/branches/*/protection/enforce_admins
- /repositories/:repository_id/branches/*/protection/required_pull_request_reviews
- /repositories/:repository_id/branches/*/protection/required_signatures
- /repositories/:repository_id/branches/*/protection/required_status_checks
- /repositories/:repository_id/branches/*/protection/required_status_checks/contexts
- /repositories/:repository_id/branches/*/protection/restrictions
- /repositories/:repository_id/branches/*/protection/restrictions/apps
- /repositories/:repository_id/branches/*/protection/restrictions/teams
- /repositories/:repository_id/branches/*/protection/restrictions/users
- /repositories/:repository_id/branches/*/push_control
- /repositories/:repository_id/branches/*/rename
- /repositories/:repository_id/check-runs
- /repositories/:repository_id/check-runs/:check_run_id
- /repositories/:repository_id/check-runs/:check_run_id/annotations
- /repositories/:repository_id/check-runs/:check_run_id/rerequest
- /repositories/:repository_id/check-suites
- /repositories/:repository_id/check-suites/:check_suite_id
- /repositories/:repository_id/check-suites/:check_suite_id/check-runs
- /repositories/:repository_id/check-suites/:check_suite_id/rerequest
- /repositories/:repository_id/check-suites/preferences
- /repositories/:repository_id/codespaces/secrets
- /repositories/:repository_id/codespaces/secrets/:name
- /repositories/:repository_id/codespaces/secrets/public-key
- /repositories/:repository_id/collaborators
- /repositories/:repository_id/collaborators/:username
- /repositories/:repository_id/collaborators/:username/permission
- /repositories/:repository_id/commits
- /repositories/:repository_id/commits/*
- /repositories/:repository_id/commits/*/branches-where-head
- /repositories/:repository_id/commits/*/check-runs
- /repositories/:repository_id/commits/*/check-suites
- /repositories/:repository_id/commits/*/pulls
- /repositories/:repository_id/contents/?*
- /repositories/:repository_id/dependabot/secrets
- /repositories/:repository_id/dependabot/secrets/:name
- /repositories/:repository_id/dependabot/secrets/public-key
- /repositories/:repository_id/downloads
- /repositories/:repository_id/downloads/:download_id
- /repositories/:repository_id/environments/:environment/deployment-branch-policies
- /repositories/:repository_id/environments/:environment/deployment-branch-policies/:branch_policy_id
- /repositories/:repository_id/git/blobs
- /repositories/:repository_id/git/blobs/:sha
- /repositories/:repository_id/git/commits
- /repositories/:repository_id/git/commits/:commit_id
- /repositories/:repository_id/git/extract-ref/*
- /repositories/:repository_id/git/matching-refs/*
- /repositories/:repository_id/git/ref/*
- /repositories/:repository_id/git/refs
- /repositories/:repository_id/git/refs/*
- /repositories/:repository_id/git/tags
- /repositories/:repository_id/git/tags/:tag_id
- /repositories/:repository_id/git/tree-file-list/*
- /repositories/:repository_id/git/trees
- /repositories/:repository_id/git/trees/*
- /repositories/:repository_id/installation
- /repositories/:repository_id/invitations
- /repositories/:repository_id/invitations/:invitation_id
- /repositories/:repository_id/license
- /repositories/:repository_id/merge-upstream
- /repositories/:repository_id/merges
- \/repositories\/(\d+)\/contents\/(.+)
- \/repositories\/(\d+)\/readme(?:\/(.*))?
- \/repositories\/(\d+)\/tarball(?:\/(.*))?
- \/repositories\/(\d+)\/zipball(?:\/(.*))?

</details>

<details>
  <summary>SCIM</summary>

- /scim/v2/enterprises/:enterprise_id/Groups
- /scim/v2/enterprises/:enterprise_id/Groups/:external_identity_guid
- /scim/v2/enterprises/:enterprise_id/Users
- /scim/v2/enterprises/:enterprise_id/Users/:external_identity_guid
- /scim/v2/organizations/:organization_id/Users
- /scim/v2/organizations/:organization_id/Users/:external_identity_guid

</details>

<details>
  <summary>Search</summary>

- /legacy/issues/search/:owner/:repo/:state/:q

</details>

**Comment by vwnj84** [comment]
We gave this a try back in March and had to shut it down as it broke audit log streaming for the enterprise. What's the current status of this beta feature? It is sorely needed especially when troubleshooting rate limiting.

**Comment by boylejj** [comment]
@vwnj84, Thanks for reaching out. I remember looking into the issues you were seeing with this feature when you initially asked us to disable it. It seemed that the increased volume exacerbated an underlying issue in your streaming configuration. Can we set up some time to troubleshoot what might be happening with this feature so you can start streaming these important events again? I can also discuss the current state of the beta with you and some changes in approach we've recently taken to minimize potential noise.

**Reply by vwnj84** [reply]
@boylejj Sure thing.

**Comment by erickatwork** [comment]
@boylejj hi, is this still in beta? If so, do you have an update on when it'll become public?

**Reply by boylejj** [reply]
👋🏻  Hey @erickatwork! Yes, this is still in beta. We are targeting this feature becoming GA next quarter. With over 1000 enterprises enrolling in the public beta, we needed further investment in scaling our streaming infra to handle the rising event traffic this rapid adoption created. Happy people really want this data, but also want to make sure it is delivered reliably.

**Reply by erickatwork** [reply]
Great thank you!
